src/base/bittorrent/session.cpp

```diff
@@ -1891,42 +1882,14 @@ bool Session::addTorrent(const MagnetUri &magnetUri, const AddTorrentParams &par
 
     const auto it = m_loadedMetadata.constFind(hash);
     if (it != m_loadedMetadata.constEnd()) {
-        // Adding preloaded torrent...
-        const TorrentInfo metadata = it->metadata;
-        if (metadata.isValid()) {
-            // Metadata is received and torrent_handle is being deleted
-            // so we can't reuse it. Just add torrent using its metadata.
-            return addTorrent(metadata, params);
-        }
-
-        // Reuse existing torrent_handle
-        lt::torrent_handle handle = m_nativeSession->find_torrent(hash);
-
-        // Preloaded torrent is in "Upload mode" so we need to disable it
-        // otherwise the torrent never be downloaded (until application restart)
-        handle.unset_flags(lt::torrent_flags::upload_mode);
-
-        LoadTorrentParams createTorrentParams = initLoadTorrentParams(params);
-        createTorrentParams.ltAddTorrentParams = it->ltAddTorrentParams;
-
-        if (createTorrentParams.paused) {
-            // Preloaded torrent isn't auto managed already
-            handle.pause();
-        }
-        else if (!createTorrentParams.forced) {
-            handle.set_flags(lt::torrent_flags::auto_managed);
-            handle.pause();
-        }
-
-        m_loadedMetadata.remove(hash);
-        m_loadingTorrents.insert(hash, createTorrentParams);
-
+        // Although it looks illogical that we just don't use an existing handle,
+        // but as previous experience has shown, it actually creates unnecessary
```

Comment:
@glassez:  While irrelevant to the actual code, the grammar in this code comment is incorrect.  "Although" and the "but" conjunction are redundant and difficult to read like this for present-day English speakers. 

Please try… 

> // **It** looks illogical that we just don't use an existing handle,
> // but as previous experience has shown,

Reply:
Thank you! My English is bad. Will fix it.

Reply:
Comment is fixed. @Chocobo1, please re-approve.